examples/unfair_dice.py

```diff
@@ -0,0 +1,199 @@
+
+        Returns:
+            True if the optimization loop should be aborted
+        """
+        self._values.append(value)
+        self._parameters.append(parameters)
+
+        if len(self._values) > self.N:
+            last_values = self._values[-self.N :]
+            pp = np.polyfit(range(self.N), last_values, 1)
+            slope = pp[0] / self.N
+
+            if slope > self.tolerance:
+                return True
+        return False
+
+
+dt = AverageDecreaseTermination(N=35)
+
+
+def U(circuit_ir, q: Qubit, theta: float, phi: float, lamb: float):
+    """McKay decomposition of the U gate.
+
+    :param self: circuit object
+    :param q: qubit
+    :param theta: angle
+    :param phi: angle
+    :param lamb: angle
+    :return: circuit object
+    """
+    circuit_ir.Rz(q, Float(phi))
+    circuit_ir.Rx(q, Float(-np.pi / 2))
+    circuit_ir.Rz(q, Float(theta))
+    circuit_ir.Rx(q, Float(np.pi / 2))
+    circuit_ir.Rz(q, Float(lamb))
+    return circuit_ir
+
+
+def generate_ansatz(params):
+    with Circuit(platform_name="spin-2", program_name="prgm1", number_of_qubits=2) as circuit:
+        U(circuit.ir, Qubit(0), *params[0:3])
+        U(circuit.ir, Qubit(1), *params[3:6])
+        circuit.ir.CZ(Qubit(0), Qubit(1))
+        U(circuit.ir, Qubit(0), *params[6:9])
+        U(circuit.ir, Qubit(1), *params[9:12])
+        for ii in range(number_of_qubits):
+            circuit.ir.measure(Qubit(ii), Bit(ii))
+
+    return circuit
+
+
+def objective_function(params, qi, target_distribution, nshots=None):
+    """Compares the output distribution of our circuit with parameters `params` to the target distribution."""
+    qc = generate_ansatz(params)
+    execute_result = qi.execute_circuit(qc.content, nshots)
+    # Convert the result to a dictionary with probabilities
+    output_distr = counts_to_distr(execute_result.results)
+    # Calculate the cost as the distance between the output
+    # distribution and the target distribution
+    cost = sum(abs(target_distribution.get(i, 0) - output_distr.get(i, 0)) for i in range(2**number_of_qubits))
+    return cost
+
+
+def data_callback(iteration: int, parameters: Any, residual: float) -> None:
+    """Callback used to store data.
+
+    Args:
+        iteration: Iteration on the optimization procedure
+        parameters: Current values of the parameters to be optimized
+        residual: Current residual (value of the objective function)
+    """
+    # do nothing for now
+    pass
+
+
+def qiskit_callback(number_evaluations, parameters, value, stepsize, accepted):
+    data_callback(number_evaluations, parameters, value)
+
+
+def execute(qi: QuantumInterface) -> None:
+    optimizer = SPSA(maxiter=100, callback=qiskit_callback, termination_checker=dt)
+
+    F = partial(objective_function, qi=qi, target_distribution=target_distribution, nshots=2000)
+
+    number_of_parameters = 12
+    initial_parameters = 0.85 * np.random.rand(
```

Comment:
This number is kinda magic

Reply:
Indeed, no idea why Pieter uses that particular value.

Reply:
Resolved because this is outside of our wheelhosue